docker/dockerfiles/be/be_entrypoint.sh

```diff
@@ -132,10 +132,21 @@ while true; do
       tail -n $nol $STARROCKS_HOME/log/be.out
   fi
 
-  # If coredump is enabled, only catch SIGABRT (6), SIGSEGV(11) and repeat launching be process,
-  # should still exit for rest error codes.
-  if [[ "$COREDUMP_ENABLED" != "true" || ($ret -ne 134 && $ret -ne 139) ]]; then
-    exit $ret
+  # Repeat launching BE process in the two scenarios:
+  #  a. DEBUG_MODE is true;
+  #  b. coredump is enabled, and the error code is SIGABRT(6) or SIGSEGV(11);
+  # otherwise BE process should still exit.
+  should_exit=true
+  if [[ "$DEBUG_MODE" == "true" ]]; then
```

Comment:
we may consider list all the logic control env variables into a central place and have comments explain each of them.

This scatter added variables makes the script harder and harder to understand and to maintain.

Reply:
agree, any place to keep the information now?

Reply:
can start with summary in comments at the top of the script.

Reply:
added comments for the ENV variables.